src/logo.tsx

```diff
@@ -0,0 +1,353 @@
+};
+
+const LogoBase: React.FC<LogoBaseProps> = ({size = 48, skinName, logoType = 'isotipo'}) => {
+    switch (skinName) {
+        case 'Movistar':
+            return <MovistarLogoImage size={size} logoType={logoType} />;
+        case 'Vivo':
+            return <VivoLogoImage size={size} logoType={logoType} />;
+        case 'O2':
+        case 'O2-classic':
+            return <O2LogoImage size={size} logoType={logoType} />;
+        case 'Telefonica':
+            return <TelefonicaLogoImage size={size} logoType={logoType} />;
+        case 'Blau':
+            return <BlauLogoImage size={size} logoType={logoType} />;
+        // todo WEB-761 what to do for unknown skinnames?
+        default:
+            return null;
+    }
+};
+type LogoPropsBase = {
+    size?: number;
+    logoType?: LogoType;
+};
+
+type LogoToProps = LogoPropsBase & {
+    trackingEvent?: TrackingEvent | ReadonlyArray<TrackingEvent>;
+    to: string;
+    fullPageOnWebView?: boolean;
+    replace?: boolean;
+};
+
+type LogoHrefProps = LogoPropsBase & {
+    trackingEvent?: TrackingEvent | ReadonlyArray<TrackingEvent>;
+    href: string;
+    newTab?: boolean;
+};
+
+type LogoOnPressProps = LogoPropsBase & {
+    trackingEvent?: TrackingEvent | ReadonlyArray<TrackingEvent>;
+    onPress: () => void;
+};
+
+type LogoProps = ExclusifyUnion<LogoPropsBase | LogoToProps | LogoHrefProps | LogoOnPressProps>;
+
+const MaybeTouchableLogo = (logoTouchableProps: LogoProps, children: React.ReactNode): JSX.Element => {
+    if (logoTouchableProps.to) {
+        return (
+            <Touchable
+                trackingEvent={logoTouchableProps.trackingEvent}
+                to={logoTouchableProps.to}
+                fullPageOnWebView={logoTouchableProps.fullPageOnWebView}
+                replace={logoTouchableProps.replace}
+            >
+                {children}
+            </Touchable>
+        );
+    }
+
+    if (logoTouchableProps.href) {
+        return (
+            <Touchable
+                trackingEvent={logoTouchableProps.trackingEvent}
+                href={logoTouchableProps.href}
+                newTab={logoTouchableProps.newTab}
+                replace={logoTouchableProps.replace}
+            >
+                {children}
+            </Touchable>
+        );
+    }
+
+    if (logoTouchableProps.onPress) {
+        return (
+            <Touchable trackingEvent={logoTouchableProps.trackingEvent} onPress={logoTouchableProps.onPress}>
+                {children}
+            </Touchable>
+        );
+    }
+
+    return <> {children}</>;
+};
+export const Logo: React.FC<LogoProps> = ({size, logoType = 'isotipo', ...props}) => {
+    const {skinName} = useTheme();
+
+    return (
+        <MaybeTouchableLogo {...props}>
+            <LogoBase skinName={skinName} logoType={logoType} size={size} />
+        </MaybeTouchableLogo>
+    );
+};
+
+export const MovistarLogo: React.FC<LogoProps> = ({size, logoType = 'isotipo', ...props}) =>
```

Comment:
I don't know I found it more compact and straightforward this way here, but maybe it's not compliant with our codebase. WDYT?